src/main/java/org/spongepowered/common/item/inventory/SpongeItemStackSnapshot.java

```diff
@@ -143,6 +144,11 @@ public int getCount() {
     }
 
     @Override
+    public boolean isEmpty() {
+        return this.itemType == ItemTypes.AIR || this.count <= 0;
```

Comment:
Are you specifically allowing air itemstacks to be created now? There is a way to validate that the `ItemStack` itself is empty (hint, look at the fields in this class).